src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -1874,6 +1872,73 @@ default void setAtIndex(ValueLayout.OfAddress layout, long index, MemorySegment
         layout.accessHandle().set(this, index * layout.byteSize(), value);
     }
 
+    /**
+     * Returns a Stream of human-readable, lines with hexadecimal values for this memory segment.
+     * <p>
+     * The exact format of the stream elements is unspecified and should not
+     * be acted upon programmatically. Loosely speaking, this method renders
+     * a format similar to the *nix command "hexdump -C".
+     * <p>
+     * As an example, a memory segment created, initialized and used as follows
+     * {@snippet lang = java:
+     * MemorySegment memorySegment = memorySession.allocate(64 + 4);
+     * memorySegment.setUtf8String(0, "The quick brown fox jumped over the lazy dog\nSecond line\t:here");
+     * memorySegment.hexStream()
+     *     .forEach(System.out::println);
+     *}
+     * might print to something like this:
+     * {@snippet lang = text:
+     * 0000000000000000  54 68 65 20 71 75 69 63  6B 20 62 72 6F 77 6E 20  |The quick brown |
+     * 0000000000000010  66 6F 78 20 6A 75 6D 70  65 64 20 6F 76 65 72 20  |fox jumped over |
+     * 0000000000000020  74 68 65 20 6C 61 7A 79  20 64 6F 67 0A 53 65 63  |the lazy dog.Sec|
+     * 0000000000000030  6F 6E 64 20 6C 69 6E 65  09 3A 68 65 72 65 00 00  |ond line.:here..|
+     * 0000000000000040  00 00 00 00                                       |....|
+     *}
+     * <p>
+     * Use a {@linkplain MemorySegment#asSlice(long, long) slice} to inspect a specific region
+     * of a memory segment.
+     *
+     * @return a Stream of human-readable, lines with hexadecimal values
+     */
+    default Stream<String> hexStream() {
```

Comment:
I think it is possible to specify the `hexDump()` method in detail and also make it more general so it can be applied to virtually any memory representation such as `byte[]` or `ByteBuffer`.

Reply:
Yep, I do think that hex dump doesn't really belong here - but should be a common functionality shared between all byte array views.

Reply:
I worked on a more comprehensive HexPrinter API that was downsized to what became java.util.HexFormat.
The HexPrinter API was sidelined into the test test library: jdk/test/lib/hexdump/HexPrinter.

I agree that support for other memory buffers/segments is desirable as a generalization, even if only to HexFormat but expands the scope beyond Panama (and could be done separately).

A more general human readable formatting/layout capability is typically beyond what's included in JDK APIs. I might look to IntelliJ or the debugger to support the inspection functionality.